src/oge/data_cleaning.py

```diff
@@ -889,6 +1025,76 @@ def manually_remove_steam_units(df):
     return df
 
 
+def identify_and_remove_steam_only_units(cems, year):
+    # get annual totals by unit
+    annual_cems = (
+        cems.groupby(["plant_id_eia", "emissions_unit_id_epa"], dropna=False)[
+            ["gross_generation_mwh", "steam_load_1000_lb", "fuel_consumed_mmbtu"]
+        ]
+        .sum()
+        .reset_index()
+    )
+
+    # merge in mapped generator ids so that we can see where the CEMS data overlaps with EIA
+    subplant_crosswalk = (
+        pd.read_csv(
+            outputs_folder(f"{year}/subplant_crosswalk_{year}.csv.zip"),
+            dtype=get_dtypes(),
+        )[["plant_id_eia", "emissions_unit_id_epa", "generator_id"]].drop_duplicates()
+    ).dropna(subset=["emissions_unit_id_epa"])
+    annual_cems = annual_cems.merge(
+        subplant_crosswalk,
+        how="left",
+        on=["plant_id_eia", "emissions_unit_id_epa"],
+        validate="1:m",
+    )
+    # raise a warning about units that have non-zero generation but are missing a
+    # generator_id. These may need to be mapped
+    potential_missing_map = annual_cems[
+        (annual_cems["generator_id"].isna()) & (annual_cems["gross_generation_mwh"] > 0)
+    ]
+    if len(potential_missing_map) > 0:
+        logger.warning(
+            "The following CEMS units report non-zero generation but are missing a mapping to an EIA generator_id"
+        )
+        logger.warning(
+            "These may need to be mapped to prevent double-counting of generation"
+        )
+        logger.warning(potential_missing_map.to_string())
+
+    # flag units that report zero generation (but non-zero fuel consumption or steam), but which have no mapping to an EIA generator
+    # we want to remove these so that we don't include these emissions in the plant
+    anomalous_units = annual_cems[
+        (annual_cems["generator_id"].isna())
+        & (annual_cems["gross_generation_mwh"] == 0)
+    ]
```

Comment:
I don't think this filter does what it is intended to do, i.e., select zero-generation and non-zero fuel consumption or steam

Reply:
Fixed, split this logic out into several functions